sdk/eventgrid/azure-eventgrid/tests/unittests/test_binary_mode.py

```diff
@@ -0,0 +1,68 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+import json
+import base64
+from azure.eventgrid._operations._patch import _to_http_request
+from azure.eventgrid._model_base import AzureJSONEncoder
+from azure.eventgrid.models import *
+from azure.core.messaging import CloudEvent
+
+class MyTestClass(object):
+    def __init__(self, name):
+        self.name = name
+    def __str__(self):
+        return self.name
+    
+class TestEGClientExceptions():
+    
+    def test_binary_request_format(self):
+        event = CloudEvent(
+            type="Contoso.Items.ItemReceived",
+            source="source",
+            subject="MySubject",
+            data=b'this is binary data',
```

Comment:
is there any type of data that _needs_ to be sent in binary_mode? or does all data sent w/ binary_mode=True also work with binary_mode=False?

non-blocking, but if there is data that doesn't work with binary_mode=False and only with True, might be good to test it here w/ binary_mode=False. And see if that raises an exception.

Reply:
all with true should work with false -- binary mode only impacts formatting request and ensures that we send in binary data